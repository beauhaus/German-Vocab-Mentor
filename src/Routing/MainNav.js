import React from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

// ReactDOM.render(<Landing db={db}/>, document.getElementById('app'));

const StyledNav = styled.nav`
    grid-row: 1;
    grid-column: 1;
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    grid-template-rows: 1fr;
    & ul {
        grid-row: 1;
        grid-column: 1/6;
        /* border: 1px solid pink; */
        color: lightblue;
        background: grey;
        display: grid;
        grid-template-columns: repeat(5, 1fr);
        grid-template-rows: 1fr;
        li {
            list-style: none;
            & a {
                color: #fff;
                font-weight: 100;
                text-decoration: none;
                float: right;
                font-size: 3rem;
                margin: .5rem;
            }
        }
        li:nth-of-type(1) {
            background: #C9e4ca;
        }
        li:nth-of-type(2) {
            background: #87bba2;
        }
        li:nth-of-type(3) {
            background: #55828b;
        }
        li:nth-of-type(4) {
            background: #3b6064;
        }
        li:nth-of-type(5) {
            background: #364958;
        }
        & > li:hover {
            box-shadow: inset 12px 0px 14px 0px rgba(0, 0, 0, 0.5);
            text-align: right;
        }

    }
    & .is-active {
        text-shadow: 1px 1px 2px #000;
        color: #fff;
    }
`;

const MainNav = (props) => (
    <StyledNav className="main-nav">
        <ul>
            <li>
                <NavLink to="/" activeClassName="is-active">Home</NavLink>
            </li>
            <li>
                <NavLink to="/template" activeClassName="is-active">Template</NavLink>
            </li>
            {/**
                <li>
                <NavLink to="/edit:0" activeClassName="is-active">Edit</NavLink>
                </li>
            */}
            <li>
                <NavLink to="/AddVocabPage" activeClassName="is-active">Add</NavLink>
            </li>
            <li>
                <NavLink to="/WordListDashboard" activeClassName="is-active">List</NavLink>
            </li>
        </ul>

    </StyledNav>
)

module.exports = MainNav;