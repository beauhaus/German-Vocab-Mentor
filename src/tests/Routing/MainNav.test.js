//react-test-renderer
import React from 'react';
import {shallow} from 'enzyme';
import MainNav from '../../Routing/MainNav'

/****** ENZYME METHODS FOR SHALLOW RENDER: ******
https://airbnb.io/enzyme/docs/api/shallow.html
************************************************/

test('should render routes correctly',() => {
    const wrapper = shallow(<MainNav/>)
    expect(wrapper).toMatchSnapshot();
})