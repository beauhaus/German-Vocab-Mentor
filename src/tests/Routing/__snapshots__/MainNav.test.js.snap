// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render routes correctly 1`] = `
<.sc-bdVaJa
  className="main-nav"
>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/template"
      >
        Template
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/AddVocabPage"
      >
        Add
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/vocabItemDashboard"
      >
        List
      </NavLink>
    </li>
  </ul>
</.sc-bdVaJa>
`;
